### DISCUSSION TITLE
Implement JS Library in custom Widget

### DISCUSSION THREAD

**Discussion by wpas81** [filamentphp/filament#9659] [Help]
Hi

I'm trying to run some javascript in a custom widget I built but for some reason it won't execute my JS-code...
I found this solution https://www.answeroverflow.com/m/1138472636779864064 and tried implementing it but somehow it won't run the script 

AdminPanelProvider.php
```
 public function boot()
    {
        FilamentAsset::register([
            Css::make('leaflet-1-9-4-css', 'https://unpkg.com/leaflet@1.9.4/dist/leaflet.css'),
            Js::make('leaflet-1-9-4-js', 'https://unpkg.com/leaflet@1.9.4/dist/leaflet.js'),            
        ]);
    }
```

visitors-heatmap-widget.blade.php
```
<x-filament-widgets::widget>
    <x-filament::section>
               <div class="h-full w-fill pt-12" id="content">
                      <div id="map" style="width: 100%; height: 100%">     </div>
            </div>
    </x-filament::section>
    <script>
        document.addEventListener('livewire:initialized', () => {
            console.log('JS started');
        })
    </script>
</x-filament-widgets::widget>
```

Any help is greatly appreciated!

**Comment by wpas81** [comment]
I found it :) Full answer here: https://stackoverflow.com/questions/77495795/leaflet-js-widget-in-filament-wont-execute

**Comment by arxjei** [comment]
If the solution doesn’t work, try the following:

- Add `protected static bool $isLazy = false;` to your widget class.
- Add `wire:ignore` to the target `<div>` or element.
- Add the `data-navigate-once` attribute to the element.
- Use a Livewire event listener for the `livewire:navigated` event.
